client/my-sites/checkout/composite-checkout/components/additional-terms-of-service-in-cart.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { TermsOfServiceRecord, useShoppingCart } from '@automattic/shopping-cart';
+import i18nCalypso, { useTranslate, TranslateResult } from 'i18n-calypso';
+import { useLocale } from '@automattic/i18n-utils';
+import { useSelector } from 'react-redux';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'calypso/components/gridicon';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+
+const debug = debugFactory( 'calypso:composite-checkout:additional-terms-of-service' );
+
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+
+export default function AdditionalTermsOfServiceInCart(): JSX.Element | null {
+	const translate = useTranslate();
+	const locale = useLocale();
+	const { responseCart } = useShoppingCart();
+	const siteSlug = useSelector( getSelectedSiteSlug );
+
+	if ( ! responseCart.terms_of_service || responseCart.terms_of_service.length === 0 ) {
+		return null;
+	}
+
+	return (
+		<>
+			{ responseCart.terms_of_service.map( ( termsOfServiceRecord, index ) => {
+				const message = getMessageForTermsOfServiceRecord(
+					termsOfServiceRecord,
+					translate,
+					locale,
+					siteSlug
+				);
+
+				if ( ! message ) {
+					return null;
+				}
+
+				return (
+					<div className="checkout__titan-terms-of-service" key={ index }>
+						<Gridicon icon="info-outline" size={ 18 } />
+						<p>{ message }</p>
+					</div>
+				);
+			} ) }
+		</>
+	);
+}
+
+function getMessageForTermsOfServiceRecord(
+	termsOfServiceRecord: TermsOfServiceRecord,
+	translate: ReturnType< typeof useTranslate >,
+	locale: string,
+	siteSlug: string | null
+): TranslateResult {
+	switch ( termsOfServiceRecord.code ) {
+		case 'terms_for_bundled_trial_auto_renewal_paypal':
+			if (
+				locale !== 'en' &&
+				! i18nCalypso.hasTranslation(
+					'At the end of the promotional period we will begin charging your PayPal account (%(email)s) the normal %(productName)s subscription price of %(renewalPrice)s. You can update the payment method at any time {{link}}here{{/link}}'
+				)
+			) {
+				return '';
+			}
+			return translate(
+				'At the end of the promotional period we will begin charging your PayPal account (%(email)s) the normal %(productName)s subscription price of %(renewalPrice)s. You can update the payment method at any time {{link}}here{{/link}}',
+				{
+					args: {
+						email: termsOfServiceRecord.args?.email,
```

Comment:
It's unlikely, and not worth spending a ton of time on, but if for some reason this data does not exist (eg: someone accidentally changes the backend without updating the code here), what's the best experience for the user?

Right now it will still show the message, but I think it will look something like `we will begin charging your PayPal account (undefined) the normal undefined subscription price of undefined`, which doesn't seem ideal.

There's different ways you could handle this. One is to set defaults, for example by changing this line to `email: termsOfServiceRecord.args?.email ?? 'unknown email address',`. 

Alternatively you could make the variables default to `''` and adjust the template string so that it still works with the data missing. So something like `we will begin charging your PayPal account%(email)s the normal%(productName)s subscription price%(renewalPrice)s` with the arg `email: termsOfServiceRecord.args?.email ? ' (${termsOfServiceRecord.args?.email})' : ''` (pretend those single quotes are backticks), etc.

Another option, and maybe the easiest, is to skip the message if the required data is missing. TypeScript should make this easy: just remove the optional chaining from these variable uses and then you'll get errors until you add them to the `if` statement above.

Reply:
I decided to skip the message, I wasn't able to think about useful default values, and the option of make them skippable in the translation also seemed too complex.